python/cudf/cudf/utils/cudautils.py

```diff
@@ -197,13 +197,15 @@ def make_cache_key(udf, sig):
     """
     codebytes = udf.__code__.co_code
     constants = udf.__code__.co_consts
+    names = udf.__code__.co_names
+
     if udf.__closure__ is not None:
         cvars = tuple(x.cell_contents for x in udf.__closure__)
         cvarbytes = dumps(cvars)
     else:
         cvarbytes = b""
 
-    return constants, codebytes, cvarbytes, sig
+    return names, constants, codebytes, cvarbytes, sig
```

Comment:
Why did the cache key change? Is that important for this PR?

Reply:
I found that this would erroneously get a cache hit unless the names of local variables were included. For instance this is what I get for `startswith` and `endswith` for the key:

```python
#(('startswith',), (None, 'str_col'), b'|\x00d\x01\x19\x00}\x01|\x01\xa0\x00\x88\x00\xa1\x01S\x00', b'\x80\x04\x95\x07\x00\x00\x00\x00\x00\x00\x00\x8c\x01a\x94\x85\x94.', (dtype('O'),))
#  (('endswith',), (None, 'str_col'), b'|\x00d\x01\x19\x00}\x01|\x01\xa0\x00\x88\x00\xa1\x01S\x00', b'\x80\x04\x95\x07\x00\x00\x00\x00\x00\x00\x00\x8c\x01a\x94\x85\x94.', (dtype('O'),))
```